cylc/flow/scripts/cylc_scan.py

```diff
@@ -60,267 +159,356 @@ def get_option_parser():
-        help="Print all available information about each suite.",
-        action="store_true", default=False, dest="full")
 
-    parser.add_option(
-        '--color', '--colour', action='store', dest='color', default='auto',
-        help='Colorize the output, can be "always", "never" or "auto".')
+def state_totals(totals, colour_blind=False):
+    """Return a string with a visual representation of a suite's state totals.
 
-    parser.add_option(
-        "--comms-timeout", metavar="SEC",
-        help="Set a timeout for network connections "
-             "to each running suite. The default is 5 seconds.",
-        action="store", default=5.0, dest="comms_timeout")
+    Args:
+        totals (dict):
+            State totals dictionary.
+        colour_blind (bool):
+            If True then we will not depend on colour to convey information.
 
-    parser.add_option(
-        '-t',
-        '--format',
-        default='plain',
-        action='store',
-        dest='format',
-        help=(
-             'Set output format:\n'
-             ' * plain (default) - text format for interactive use\n'
-             ' * raw - parsable format (suite|owner|host|property|value)\n'
-             ' * json - JSON format ({suite: {owner: OWNER, host: HOST ...)')
+    Returns:
+        str
+
+    """
+    ret = []
+    for state, tag in JOB_COLOURS.items():
+        number = totals[state]
+        if number == 0:
+            continue
+        if colour_blind:
+            ret.append(f'<{tag}>{state}:{number}</{tag}>')
+        else:
+            ret.append(f'<{tag}>{number} ■</{tag}>')
+    return ' '.join(ret)
+
+
+def state_totals_key():
+    """Return a key to accompany state_totals."""
+    return '<b>Job State Key: </b>' + ' '.join(
+        f'<{tag}>\u25A0 {name}</{tag}>'
+        for name, tag in JOB_COLOURS.items()
     )
 
-    return parser
 
+def _format_json(items, _):
+    """A JSON formatter."""
+    return json.dumps(
+        [
+            {
+                key: str(value) if isinstance(value, Path) else value
+                for key, value in flow.items()
+            }
+            for flow in items
+        ],
+        indent=4,
+        sort_keys=True
+    )
 
-@terminal.cli_function(get_option_parser)
-def main(parser, options):
-    """Implement "cylc scan"."""
-    if options.full:
-        options.describe = options.state_totals = options.publisher = True
-    if options.format in ['raw', 'json']:
-        options.color = False
 
-    # color settings
-    if options.color in ['auto', 'always'] and terminal.supports_color():
-        options.color = True
+def _format_plain(flow, _):
+    """A single line format of the form: <name> [<host>:<port>]"""
+    if flow.get('contact'):
+        return f'<b>{flow["name"]}</b> {flow[Cont.HOST]}:{flow[Cont.PORT]}'
     else:
-        options.color = False
-    color_init(autoreset=True, strip=not options.color)
+        return f'<dim><b>{flow["name"]}</b></dim>'
```

Comment:
Is this a html tag? Should it be div?
```suggestion
        return f'<div><b>{flow["name"]}</b></div>'
```

Reply:
This is an `ansimarkup` tag, `dim` is term speak for "diminished".

Reply:
Ah good to know!